Entities/SceneLayer.h

```diff
@@ -316,8 +346,50 @@ namespace RTE {
 		void Clear();
 
 		// Disallow the use of some implicit methods.
-		SceneLayer(const SceneLayer &reference) = delete;
-		void operator=(const SceneLayer &rhs) = delete;
+		SceneLayerImpl(const SceneLayerImpl&reference) = delete;
+		void operator=(const SceneLayerImpl&rhs) = delete;
+	};
+
+	// If we track drawings, then disallow getting non-const access to the underlying bitmap - we must draw through special functions on scenelayer that'll track the drawings
+	class SceneLayerTracked : public SceneLayerImpl<true> {
```

Comment:
I don't think I like this template version vs. just standard inheritance. It seems to me that it's the sort of clever code that makes things more complex than they need to be.

That said, I'm not gonna make a fuss about it, it works fine and it's clear enough when one reads through it, so it's fine to stay as-is.

Reply:
It’s to avoid needing a whole shitload of duplicate code. Standard inheritance would require either copying two versions of the function, one with and one without the registered stuff, or it would require having a bug of runtime checks to see if it’s a tracked draw or not.

I do concur that it’s rather ugly, though. Multithreaded sim and render removes the ColorMO layer altogether, so this code becomes unused. I may adapt it in future to allow cheap copies so we can more efficiently pass around scene-wide maps, but it depends on how badly I want to avoid the scene layers changing mid-render. To be honest it probably isn’t a big issue anyways.

Reply:
Yeah, I think you're right about duplicates if you had some base class that SceneLayer and SceneLayerTracked both inherited.
I guess you could make SceneLayerTracked inherit SceneLayer and override methods without calling the base class methods, but that's also wonky.

Anyway, we agree it's kinda ugly but it's okay haha, and if you're removing it or fiddling with it in future, that may solve the ugliness.